jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/AllowedResourceAliasChecker.java

```diff
@@ -172,14 +180,20 @@ protected boolean isAllowed(Path path)
             {
                 // If the path is the same file as the base, then it is contained in the base and
                 // is not protected.
-                if (isSameFile(path, _base))
+                if (isSameFile(path, _baseResource))
                     return true;
 
                 // If the path is the same file as any protected resources, then it is protected.
-                for (Path p : _protected)
+                for (String protectedTarget : _protected)
                 {
-                    if (isSameFile(path, p))
-                        return false;
+                    Resource p = _baseResource.resolve(protectedTarget);
```

Comment:
Don't we already have the precise Resource already resolved and handed to us in the `public boolean checkAlias(String pathInContext, Resource resource)` method?

Why are we attempting to resolve it again against the baseResource?

Reply:
We're resolving the protectedTarget here. We can pull a list of protectedTargets as strings from the `ContextHandler`.

Previously we were resolving the protectedTarget as a path against the base resource during the initialization. However with the `CombinedResource` there may be multiple protected targets for each string like `WEB-INF`. If we resolve them at initialization they could be null meaning they don't exist at that time.

Reply:
@lachlan-roberts Could you do the resolves of protected resources to resources in `extraceBaseResourceFromContext` and then just have a list of them (yes they would now be a `List<Resource>` rather than a `CombinedResource` that contained a `List<Resources>`)

We can have Resources that don't exist right???

Reply:
> We can have Resources that don't exist right???

No we can't, if you do the `resource.resolve()` and it doesn't exist it will return null.

Reply:
I really do not like the change in resource that made resolve return null for a non existent resource.  It makes doing logic on resources too difficult.

We should look at reverting that... but in a different PR. @lachlan-roberts can you open an issue for that and then resolve this conversation.

Reply:
I opened https://github.com/jetty/jetty.project/issues/11411